src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -348,6 +348,32 @@ private static object GetTypeAsDiagnosticArgument(TypeSymbol typeOpt)
             return typeOpt ?? (object)"<null>";
         }
 
+        private bool ReportNullReferenceAssignmentIfNecessary(BoundExpression value, TypeSymbolWithAnnotations targetType, TypeSymbolWithAnnotations valueType, bool useLegacyWarnings)
```

Comment:
>ReportNullReferenceAssignmentIfNecessary [](start = 21, length = 40)

It would be good to add a comment explaining how to interpret return value for this function. For example, it looks like it can return true even if no warnings were reported. It would also be good to document other similar functions like ```CheckNullAsNonNullableReference```, etc.

Reply:
The method should only return `true` if a warning has been reported.

Reply:
> The method should only return true if a warning has been reported. 

It looks like ```CheckNullAsNonNullableReference``` can return true without reporting anything, then this function returns true.


---
In reply to: [182304386](https://github.com/dotnet/roslyn/pull/26107#discussion_r182304386) [](ancestors = 182304386)

Reply:
Removed option from `CheckNullAsNonNullableReference`.

---
In reply to: [182310734](https://github.com/dotnet/roslyn/pull/26107#discussion_r182310734) [](ancestors = 182310734,182304386)